beacon-chain/rpc/eth/validator/handlers_block.go

```diff
@@ -21,7 +21,119 @@ import (
+func (s *Server) ProduceBlockV2(w http.ResponseWriter, r *http.Request) {
+	ctx, span := trace.StartSpan(r.Context(), "validator.ProduceBlockV2")
+	defer span.End()
+
+	if shared.IsSyncing(ctx, w, s.SyncChecker, s.HeadFetcher, s.TimeFetcher, s.OptimisticModeFetcher) {
+		return
+	}
+
+	segments := strings.Split(r.URL.Path, "/")
+	rawSlot := segments[len(segments)-1]
+	rawRandaoReveal := r.URL.Query().Get("randao_reveal")
+	rawGraffiti := r.URL.Query().Get("graffiti")
+	rawSkipRandaoVerification := r.URL.Query().Get("skip_randao_verification")
+
+	slot, valid := shared.ValidateUint(w, "slot", rawSlot)
+	if !valid {
+		return
+	}
+
+	var randaoReveal []byte
+	if rawSkipRandaoVerification == "true" {
+		randaoReveal = primitives.PointAtInfinity
+	} else {
+		rr, err := shared.DecodeHexWithLength(rawRandaoReveal, fieldparams.BLSSignatureLength)
+		if err != nil {
+			http2.HandleError(w, errors.Wrap(err, "unable to decode randao reveal").Error(), http.StatusBadRequest)
+			return
+		}
+		randaoReveal = rr
+	}
+	var graffiti []byte
+	if rawGraffiti != "" {
+		g, err := shared.DecodeHexWithLength(rawGraffiti, 32)
+		if err != nil {
+			http2.HandleError(w, errors.Wrap(err, "unable to decode graffiti").Error(), http.StatusBadRequest)
+			return
+		}
+		graffiti = g
+	}
+
+	s.produceBlockV3(ctx, w, r, &eth.BlockRequest{
+		Slot:         primitives.Slot(slot),
+		RandaoReveal: randaoReveal,
+		Graffiti:     graffiti,
+		SkipMevBoost: false,
+	}, full)
+}
+
+// DEPRECATED: Please use ProduceBlockV3 instead.
+//
+// ProduceBlindedBlock requests the beacon node to produce a valid unsigned blinded beacon block,
+// which can then be signed by a proposer and submitted.
+func (s *Server) ProduceBlindedBlock(w http.ResponseWriter, r *http.Request) {
+	ctx, span := trace.StartSpan(r.Context(), "validator.ProduceBlindedBlock")
+	defer span.End()
+
+	if shared.IsSyncing(ctx, w, s.SyncChecker, s.HeadFetcher, s.TimeFetcher, s.OptimisticModeFetcher) {
+		return
+	}
+
+	segments := strings.Split(r.URL.Path, "/")
+	rawSlot := segments[len(segments)-1]
+	rawRandaoReveal := r.URL.Query().Get("randao_reveal")
+	rawGraffiti := r.URL.Query().Get("graffiti")
+	rawSkipRandaoVerification := r.URL.Query().Get("skip_randao_verification")
+
+	slot, valid := shared.ValidateUint(w, "slot", rawSlot)
+	if !valid {
+		return
+	}
+
+	var randaoReveal []byte
+	if rawSkipRandaoVerification == "true" {
+		randaoReveal = primitives.PointAtInfinity
+	} else {
+		rr, err := shared.DecodeHexWithLength(rawRandaoReveal, fieldparams.BLSSignatureLength)
+		if err != nil {
+			http2.HandleError(w, errors.Wrap(err, "unable to decode randao reveal").Error(), http.StatusBadRequest)
+			return
+		}
+		randaoReveal = rr
+	}
+	var graffiti []byte
+	if rawGraffiti != "" {
+		g, err := shared.DecodeHexWithLength(rawGraffiti, 32)
+		if err != nil {
+			http2.HandleError(w, errors.Wrap(err, "unable to decode graffiti").Error(), http.StatusBadRequest)
+			return
+		}
+		graffiti = g
+	}
```

Comment:
At a glance it looks like these 3 functions are doing the same thing for this portion of code, to avoid duplication we should consider extracting it into it's own function

Reply:
I was thinking about this, but the other two methods are marked as deprecated and will be removed eventually, so I don't feel to bad copy-pasting the code.